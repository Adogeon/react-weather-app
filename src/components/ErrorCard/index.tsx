import type { rawAPIError } from "../../types/api";
import styles from "./ErrorCard.module.css";

const ErrorCard = ({ data }: { data: rawAPIError }) => (
  <section className={styles["error"]}>
    <h1>{data.code}</h1>
    <p>{data.message}</p>
  </section>
);

export default ErrorCard;
