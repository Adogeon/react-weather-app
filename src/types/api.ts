import type { CurrentWeather, DailyForecastWithHour } from "./weather";

export interface rawAPIError {
    error: {
        code: string;
        message: string;
    };
}

export interface rawAPISuccess {
    location: {
        name: string;
        region: string;
        country: string;
    };
    current: CurrentWeather
    forecast: {
        forecastday: Array<DailyForecastWithHour>
    }
};
