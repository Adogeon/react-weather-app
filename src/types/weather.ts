export interface WeatherCondition {
    text: string;
    icon: string;
}

export interface TempAndConditions {
    temp_c: string;
    temp_f: string;
    feelslike_c: string;
    feelslike_f: string;
    condition: WeatherCondition;
}

export interface CurrentWeather extends TempAndConditions {
    last_updated: string
}

interface DailyForecastWeather {
    maxtemp_c: string;
    maxtemp_f: string;
    mintemp_c: string;
    mintemp_f: string;
    daily_chance_of_rain: string;
    avghumidity: string;
    condition: WeatherCondition;
}

export interface DailyForecastData {
    date: string;
    day: DailyForecastWeather
}

interface HourlyForecast extends TempAndConditions {
    time: string;
}

export interface DailyForecastWithHour extends DailyForecastData {
    hour: Array<HourlyForecast>
}