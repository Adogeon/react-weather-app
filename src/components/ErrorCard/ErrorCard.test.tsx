import { test, expect } from "vitest";
import { render } from "@testing-library/react";
import ErrorCard from ".";
import { rawAPIError } from "../../types/api";

const testData: rawAPIError = {
  error: {
    code: "500",
    message: "Failed to fetch weather data",
  },
};

test("ErrorCard should render correctly", () => {
  const { asFragment } = render(<ErrorCard data={testData} />);

  expect(asFragment()).toMatchSnapshot();
});
