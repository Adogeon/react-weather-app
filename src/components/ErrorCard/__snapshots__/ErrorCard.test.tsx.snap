// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ErrorCard should render correctly 1`] = `
<DocumentFragment>
  <section
    class="_error_286e5e"
  >
    <h1>
      500
    </h1>
    <p>
      Failed to fetch weather data
    </p>
  </section>
</DocumentFragment>
`;
